### PR TITLE
CBL-5390: Add parentObj to class Logging

### PR DESCRIPTION
**Pull request by jianminzhao**
- Logging::setParentObjectRef(parentObjRef)
- For objects derived from Worker, set parent object according to the parent relationship of Worker.
- Applied parent object for a non-Worker (Connection) to Replicator.
- Includes day in the timestamp.
- Removed brackes of domain tag in the log, namely [Sync] to Sync.
- Disabled several tests that check the logs. They will be reinstated at the end of logging enhancement effort.

### CODE REVIEW

**Comment by jianminzhao** [comment]
Examples:
2024-02-23T18:43:18.062004 Sync Obj=/Repl#167/inserter#169/ {Coll#0} Inserted   6 revs in  49.26ms (  122/sec) of which 65.2% was commit
2024-02-23T18:43:18.244072 BLIP Obj=/Repl#164/Connection#163/ Closing with code 1000, msg ''
(Connection is a Worker)
2024-02-23T18:43:18.783885 Sync Obj=/Repl#210/pusher#211/ {Coll#0} Peer is pulling changes from seq #0

**Comment by borrrden** [comment]
Is this UTC time?  If so is there supposed to be a Z at the end of the timestamp?

File: LiteCore/Support/LogEncoder.cc [hunk 1]
```diff
@@ -99,7 +99,7 @@ namespace litecore {
             if ( i == objectMap.end() ) {
                 _writer.write({"?\0", 2});
             } else {
-                _writer.write(slice(i->second.c_str()));
+                _writer.write(slice(i->second.first.c_str()));
```
Comment (borrrden) [inline LiteCore/Support/LogEncoder.cc:102]: `second.first` is a hard to read code smell

**Review by borrrden — APPROVED** [review]
The effect is nice but the whole map inside a map thing decreases readability

**Review by pasin — APPROVED** [review]
`/Repl#167/inserter#169` : Prefer to have naming format consistency (case wise). This can be done as a separate task and not high priority.

**Pull request merged**